test/unittests/state_deposit_requests_test.cpp

```diff
@@ -0,0 +1,33 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2025 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <gtest/gtest.h>
+#include <test/state/requests.hpp>
+
+using namespace evmc::literals;
+using namespace evmone::state;
+
+TEST(state_deposit_requests, collect_deposit_requests)
+{
+    std::array receipts{
+        TransactionReceipt{.logs = {Log{.addr = DEPOSIT_CONTRACT_ADDRESS}}},
+    };
+
+    auto& log_data = receipts[0].logs[0].data;
+    log_data = bytes(576, 0xfe);              // fill expected length with 0xfe
+    log_data.replace(6 * 32, 48, 48, 0x01);   // pubkey
+    log_data.replace(9 * 32, 32, 32, 0x02);   // withdrawal_credentials
+    log_data.replace(11 * 32, 8, 8, 0x03);    // amount
+    log_data.replace(13 * 32, 96, 96, 0x04);  // signature
+    log_data.replace(17 * 32, 8, 8, 0x05);    // index
+
+    const auto requests = collect_deposit_requests(receipts);
+    EXPECT_EQ(requests.type(), Requests::Type::deposit);
+    EXPECT_EQ(requests.data().size(), 192);
+    EXPECT_EQ(requests.data().substr(0, 48), bytes(48, 0x01));
```

Comment:
could be fewer offset calculations like
```suggestion
    EXPECT_EQ(requests.data(), bytes(48, 0x01) + bytes(32, 0x02) + ...);
```

Reply:
Make sense. I will change it.